packages/typespec-client-generator-core/src/http.ts

```diff
@@ -624,26 +654,27 @@ export function getCorrespondingMethodParams(
   serviceParam.onClient = false;
 
   // 4. To see if the service parameter is a method parameter or a property of a method parameter.
-  const directMapping = findMapping(context, methodParameters, serviceParam);
-  if (directMapping) {
-    return diagnostics.wrap([directMapping]);
+  const directMappingPath = findMappingWithPath(context, methodParameters, serviceParam);
+  if (directMappingPath) {
+    return diagnostics.wrap([directMappingPath]);
   }
 
   // 5. To see if all the property of the service parameter could be mapped to a method parameter or a property of a method parameter.
+  // This is the spread body case where multiple paths may exist.
   if (serviceParam.kind === "body" && serviceParam.type.kind === "model") {
-    const retVal = [];
+    const paths: (SdkMethodParameter | SdkModelPropertyType)[][] = [];
     let optionalSkip = 0;
     for (const serviceParamProp of serviceParam.type.properties) {
-      const propertyMapping = findMapping(context, methodParameters, serviceParamProp);
-      if (propertyMapping) {
-        retVal.push(propertyMapping);
+      const propertyMappingPath = findMappingWithPath(context, methodParameters, serviceParamProp);
+      if (propertyMappingPath) {
+        paths.push(propertyMappingPath);
       } else if (serviceParamProp.optional) {
         // If the property is optional, we can skip the mapping.
         optionalSkip++;
```

Comment:
Hard for me to reason why this be skipped when the prop is optional.

But not related to this PR.